posthog/tasks/process_scheduled_changes.py

```diff
@@ -59,6 +61,17 @@ def is_unrecoverable_error(exception: Exception) -> bool:
     return isinstance(exception, unrecoverable_types)
 
 
+def compute_next_run(current: datetime, interval: str) -> datetime:
+    """Compute the next scheduled run time based on recurrence interval."""
+    if interval == "daily":
+        return current + relativedelta(days=1)
+    elif interval == "weekly":
+        return current + relativedelta(weeks=1)
+    elif interval == "monthly":
+        return current + relativedelta(months=1)
+    raise ValueError(f"Unknown recurrence interval: {interval}")
```

Comment:
Using `relativedelta(months=1)` handles most cases, but consider documenting or testing behavior for edge cases like scheduling on Jan 31st (would go to Feb 28/29).